python/veles/db/tracker.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2017 CodiLime
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import weakref
+
+try:
+    from functools import lru_cache
+except ImportError:
+    from backports.functools_lru_cache import lru_cache
+
+from veles.schema.nodeid import NodeID
+from veles.proto import operation, check
+from veles.proto.node import Node, PosFilter
+from veles.proto.exceptions import (
+    ObjectGoneError,
+    ObjectExistsError,
+    ParentCycleError,
+    PreconditionFailedError,
+)
+
+from .subscriber import (
+    BaseSubscriberNode,
+    BaseSubscriberData,
+    BaseSubscriberBinData,
+    BaseSubscriberList,
+)
+from .backend import DbBackend
+from .transaction import Transaction
+from .node import DbNode
+
+
+class DbTracker(object):
+    def __init__(self, db):
+        if not isinstance(db, DbBackend):
```

Comment:
wut? You allow for someone to pass something else than DbBackend instance, and then silently fix it?